x-pack/plugin/index-lifecycle/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleService.java

```diff
@@ -44,18 +40,15 @@
     private final PolicyStepsRegistry policyRegistry;
     private Client client;
     private ClusterService clusterService;
-    private ThreadPool threadPool;
     private LongSupplier nowSupplier;
     private SchedulerEngine.Job scheduledJob;
     private IndexLifecycleRunner lifecycleRunner;
 
-    public IndexLifecycleService(Settings settings, Client client, ClusterService clusterService, Clock clock,
```

Comment:
ThreadPool is no longer used since the service does not issue a cluster-state-update to install metadata anymore